client/python/datajunction/client.py

```diff
@@ -463,35 +475,89 @@ def sql(  # pylint: disable=too-many-arguments
                 "metrics": metrics,
                 "dimensions": dimensions,
                 "filters": filters,
-                "engine_name": engine_name,
-                "engine_version": engine_version,
+                "engine_name": engine_name or self.engine_name,
+                "engine_version": engine_version or self.engine_version,
             },
         )
         if response.status_code == 200:
             return response.json()["sql"]
         return response.json()
 
-    def data(  # pylint: disable=too-many-arguments
+    def data_for_single_node(  # pylint: disable=too-many-arguments
```

Comment:
I'm starting to think that we should actually limit the data retrieval in the client to dimensions, metrics, and cubes. We can perhaps offer a sampling endpoint for other nodes (and of course the API itself still exists). Because otherwise the data retrieved for the intermediate nodes could be very large and I'm not sure we want the burden of operating a generic querying service.

Reply:
Fully agree, great point. so sounds like I should just remove this method entirely and have data access be through the `DJClient.data` method? I can make that method's args more flexible, for example allowing the `metrics` arg type to be `List[str | Metric]`.